### PR TITLE
Add accounting policies service and transformer

### PR DESCRIPTION
**Pull request by BSaunders95**
Add:
- `AccountingPoliciesService` 
- `AccountingPoliciesTransformer`
- `BasisOfPreparationService`

And supporting models and tests

Required for SFA-628

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/AccountingPoliciesService.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.companieshouse.web.accounts.service.smallfull;
+
+import java.util.List;
+import uk.gov.companieshouse.api.model.accounts.smallfull.AccountingPoliciesApi;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+public interface AccountingPoliciesService {
+
+    AccountingPoliciesApi getAccountingPoliciesApi(String transactionId, String companyAccountsId)
```
Comment (Mathsh) [inline src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/AccountingPoliciesService.java]: Thought the rules was to have JavaDocs on all public methods?

File: src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    @DisplayName("Update accounting policies - bad request")
+    void updateAccountingPoliciesBadRequest()
+            throws ApiErrorResponseException, URIValidationException, ServiceException {
+
+        AccountingPoliciesApi accountingPoliciesApi = new AccountingPoliciesApi();
+
+        when(accountingPoliciesResourceHandler.update(ACCOUNTING_POLICIES_URI, accountingPoliciesApi))
+                .thenReturn(accountingPoliciesUpdate);
+
+        HttpResponseException httpResponseException = new HttpResponseException.Builder(400,"Bad Request", new HttpHeaders()).build();
```
[the remark this answers is not in the document]
Comment (BSaunders95) [reply src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java]: Done 👍

File: src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    @DisplayName("Create accounting policies - bad request without validation errors")
+    void createAccountingPoliciesBadRequestWithoutValidationErrors()
+            throws ApiErrorResponseException, URIValidationException {
+
+        AccountingPoliciesApi accountingPoliciesApi = new AccountingPoliciesApi();
+
+        when(accountingPoliciesResourceHandler.create(ACCOUNTING_POLICIES_URI, accountingPoliciesApi))
+                .thenReturn(accountingPoliciesCreate);
+
+        HttpResponseException httpResponseException = new HttpResponseException.Builder(400,"Bad Request", new HttpHeaders()).build();
```
[the remark this answers is not in the document]
Comment (BSaunders95) [reply src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java]: Done 👍

File: src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    @DisplayName("Create accounting policies - bad request")
+    void createAccountingPoliciesBadRequest()
+            throws ApiErrorResponseException, URIValidationException, ServiceException {
+
+        AccountingPoliciesApi accountingPoliciesApi = new AccountingPoliciesApi();
+
+        when(accountingPoliciesResourceHandler.create(ACCOUNTING_POLICIES_URI, accountingPoliciesApi))
+                .thenReturn(accountingPoliciesCreate);
+
+        HttpResponseException httpResponseException = new HttpResponseException.Builder(400,"Bad Request", new HttpHeaders()).build();
```
[the remark this answers is not in the document]
Comment (BSaunders95) [reply src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java]: Done 👍

File: src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+    }
+
+    @Test
+    @DisplayName("Get accounting policies - resource not found")
+    void getAccountingPoliciesResourceNotFound()
+            throws ApiErrorResponseException, URIValidationException, ServiceException {
+
+        when(accountingPoliciesResourceHandler.get(ACCOUNTING_POLICIES_URI)).thenReturn(accountingPoliciesGet);
+
+        HttpResponseException httpResponseException = new HttpResponseException.Builder(404,"Not found", new HttpHeaders()).build();
```
[the remark this answers is not in the document]
Comment (BSaunders95) [reply src/test/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/AccountingPoliciesServiceImplTests.java]: Done 👍

**Comment by BSaunders95** [comment]
Javadocs added

**Pull request merged**